provider/core/src/marker/macros.rs

```diff
@@ -13,48 +13,20 @@
 /// The data struct must have a covariant lifetime.
 #[macro_export]
 macro_rules! unsafe_impl_data_marker_with_lifetime {
```

Comment:
Nit: This macro no longer hides any `unsafe` code, so please remove `unsafe_` from the name. :)